dask/blockwise.py

```diff
@@ -662,12 +724,24 @@ def make_blockwise_graph(func, output, out_indices, *arrind_pairs, **kwargs):
     deserializing = kwargs.pop("deserializing", False)
     func_future_args = kwargs.pop("func_future_args", None)
     return_key_deps = kwargs.pop("return_key_deps", False)
+    io_deps = kwargs.pop("io_deps", {})
     if return_key_deps:
         key_deps = {}
 
     if deserializing:
         from distributed.worker import warn_dumps, dumps_function
 
+    # Check if there are "func" arguments in `io_deps`.  If so,
+    # we store the real (registered) function in `io_dep_funcs`.
+    io_dep_funcs = {}
+    for arg, val in io_deps.items():
+        if val and "func" in val:
+            if val["func"][0] not in blockwise_io_reg:
+                raise RuntimeError(
+                    f"{val['func'][0]} is not registered in blockwise_io_reg."
+                )
+            io_dep_funcs[arg] = blockwise_io_reg.get(val["func"][0])
```

Comment:
```suggestion
        try:
            funcname = val["func"][0]
        except KeyError:
            continue
        try:
            io_dep_funcs[arg] = blockwise_io_reg[funcname]
        except KeyError:
            raise RuntimeError(f"{funcname} is not registered in blockwise_io_reg")
```

Reply:
although I wonder do we really need to bother with the second try... except block? Is there an added benefit to just raising a generic KeyError?

Reply:
not sure if ``val`` can be None? In that case the first one becomes ``except (KeyError, TypeError)``

Reply:
`val` cannot be None, and I agree that it probably makes the most sense to allow the generic KeyError if the funciton is not properly registered.